### PR TITLE
[ROCm] amax hipblaslt integration

### PR DESCRIPTION
**Pull request by alugorey**
AMAX is coming as part of rocm6.2. This code adds that functionality

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

File: aten/src/ATen/cuda/CUDABlas.cpp [hunk 1]
```diff
@@ -1485,10 +1485,13 @@ void scaled_gemm(
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_A_SCALE_POINTER, mat1_scale_ptr);
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_B_SCALE_POINTER, mat2_scale_ptr);
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_D_SCALE_POINTER, result_scale_ptr);
+#if defined(USE_ROCM) && ROCM_VERSION >= 60200
```
Comment (eqy) [inline aten/src/ATen/cuda/CUDABlas.cpp:1488]: Does this preserve the semantics of the non-ROCM use-case?

Comment (alugorey) [reply aten/src/ATen/cuda/CUDABlas.cpp]: Nope! My mistake, leftover from debug. I will fix

**Review by eqy — APPROVED** [review]

**Comment by alugorey** [comment]
@malfet @lezcano 
Hello! can we land this?

**Review by lezcano — APPROVED** [review]
Approving. @eqy already reviewed.

**Pull request closed without merging**